src/views/CreateEditHolder.js

```diff
@@ -2,76 +2,74 @@ import React, { Component } from "react";
 import { connect } from "react-redux";
 import PropTypes from "prop-types";
 import RichTextEditor from "react-rte";
-import Navbar from "../components/common/AppBars/navBar";
 import Validator from "../utils/validator";
+import { checkNextProps } from "../utils/helperFunctions";
 import Input from "../components/common/Inputs/TextInput";
 import Button from "../components/common/Buttons/BasicButton";
-import Success from "../components/common/Message/success";
 import Error from "../components/common/Message/error";
 import Tag from "../components/common/Tags/NewTag";
 import { toolbarConfig } from "../utils/editor/editorConfig";
 import {
-  handleInputField,
   editArticle,
-  handleCreateTag,
-  fetchOneArticle,
-  removeTag
+  newArticle,
+  fetchOneArticle
 } from "../redux/actions/newArticle";
 
 export const mapDispatchToProps = dispatch => ({
-  onInputChange: field => dispatch(handleInputField(field)),
   fetchArticleToEdit: slug => dispatch(fetchOneArticle(slug)),
-  newTag: tag => dispatch(handleCreateTag(tag)),
-  editOneArticle: (article, slug) => dispatch(editArticle(article, slug)),
-  handleRemoveTag: tag => dispatch(removeTag(tag))
+  createArticle: (article, history) => dispatch(newArticle(article, history)),
+  editOneArticle: (article, slug, history) =>
+    dispatch(editArticle(article, slug, history))
 });
 
-export const mapStateToProps = (state, ownProps) => ({
-  response: state.article.response,
-  article: state.article,
-  ...ownProps
+export const mapStateToProps = state => ({
+  ...state.newArticle,
+  ...state.auth
 });
 
-export class EditArticle extends Component {
+export class Article extends Component {
   constructor(props) {
     super(props);
-
     this.state = {
+      edit: false,
       tag: "",
-      title: props.article.title,
+      title: "",
       isFetching: false,
-      description: props.article.description,
-      tagsList: props.article.tagsList,
+      description: "",
+      tagsList: [],
       error: "",
-      value: RichTextEditor.createValueFromString(props.article.body, "html")
+      value: RichTextEditor.createEmptyValue()
     };
   }
 
   componentDidMount = () => {
-    const { match, fetchArticleToEdit } = this.props;
-    const { slug } = match.params;
-    fetchArticleToEdit(slug);
+    const {
+      match: {
+        params: { slug }
+      },
+      fetchArticleToEdit,
+      history: {
+        location: { pathname }
+      }
+    } = this.props;
+    if (pathname !== "/articles/new") {
+      this.setState({ edit: true, isFetching: true });
```

Comment:
Why are you accessing state directly here?
Have you tried all your best to use redux and it didn't work?

Reply:
Redux can be used, but again not over using the redux store. These fields from this particular component are only being used by this particular component. Then why making them available to other components which don't need them? Therefore, they have to be controlled in component which needs them only.